Dockerfile

```diff
@@ -205,6 +205,9 @@ RUN TARGETARCH_SYNONYM=$([[ "$TARGETARCH" == "amd64" ]] && echo "x86_64" || echo
     ( rm -rf $ES_BASE_DIR/modules/x-pack-ml/platform && \
         rm -rf $ES_BASE_DIR/modules/ingest-geoip)
 
+# TODO: Refactor it into a transcribe dependency installer
+# install ffmpeg package to support multiple audio formats transcription
+RUN apt-get update && apt-get install -y ffmpeg
```

Comment:
Unfortunately, this is not an option (not even temporary). This drastically increases the Docker image size:
```
root@89c127bd5e65:/opt/code/localstack# apt install ffmpeg
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  dbus fontconfig fontconfig-config fonts-dejavu-core i965-va-driver intel-media-va-driver libaacs0 libaom0 libapparmor1 libasound2 libasound2-data libass9 libasyncns0 libavc1394-0 libavcodec58 libavdevice58 libavfilter7 libavformat58 libavresample4 libavutil56
  libbdplus0 libbluray2 libbs2b0 libcaca0 libcairo2 libcdio-cdda2 libcdio-paranoia2 libcdio18 libchromaprint1 libcodec2-0.8.1 libcroco3 libcrystalhd3 libdatrie1 libdbus-1-3 libdc1394-22 libdrm-amdgpu1 libdrm-common libdrm-intel1 libdrm-nouveau2 libdrm-radeon1 libdrm2
  libelf1 libfftw3-double3 libflac8 libflite1 libfontconfig1 libfreetype6 libfribidi0 libgdk-pixbuf2.0-0 libgdk-pixbuf2.0-bin libgdk-pixbuf2.0-common libgl1 libgl1-mesa-dri libglapi-mesa libglib2.0-0 libglib2.0-data libglvnd0 libglx-mesa0 libglx0 libgme0 libgomp1
  libgraphite2-3 libgsm1 libharfbuzz0b libice6 libiec61883-0 libigdgmm5 libjack-jackd2-0 libjbig0 libjpeg62-turbo liblilv-0-0 libmp3lame0 libmpg123-0 libmysofa0 libnorm1 libnuma1 libogg0 libopenal-data libopenal1 libopenjp2-7 libopenmpt0 libopus0 libpango-1.0-0
  libpangocairo-1.0-0 libpangoft2-1.0-0 libpciaccess0 libpgm-5.2-0 libpixman-1-0 libpng16-16 libpostproc55 libpulse0 libraw1394-11 librsvg2-2 librsvg2-common librubberband2 libsamplerate0 libsdl2-2.0-0 libsensors-config libsensors5 libserd-0-0 libshine3 libslang2
  libsm6 libsnappy1v5 libsndfile1 libsndio7.0 libsodium23 libsord-0-0 libsoxr0 libspeex1 libsratom-0-0 libssh-gcrypt-4 libswresample3 libswscale5 libthai-data libthai0 libtheora0 libtiff5 libtwolame0 libusb-1.0-0 libva-drm2 libva-x11-2 libva2 libvdpau-va-gl1
  libvdpau1 libvidstab1.1 libvorbis0a libvorbisenc2 libvorbisfile3 libvpx5 libwavpack1 libwayland-client0 libwayland-cursor0 libwayland-egl1 libwebp6 libwebpmux3 libx11-6 libx11-data libx11-xcb1 libx264-155 libx265-165 libxau6 libxcb-dri2-0 libxcb-dri3-0 libxcb-glx0
  libxcb-present0 libxcb-render0 libxcb-shape0 libxcb-shm0 libxcb-sync1 libxcb-xfixes0 libxcb1 libxcursor1 libxdamage1 libxdmcp6 libxext6 libxfixes3 libxi6 libxinerama1 libxkbcommon0 libxrandr2 libxrender1 libxshmfence1 libxss1 libxtst6 libxv1 libxvidcore4
  libxxf86vm1 libzmq5 libzvbi-common libzvbi0 mesa-va-drivers mesa-vdpau-drivers shared-mime-info va-driver-all vdpau-driver-all x11-common xdg-user-dirs xkb-data
Suggested packages:
  default-dbus-session-bus | dbus-session-bus ffmpeg-doc i965-va-driver-shaders libasound2-plugins alsa-utils libbluray-bdj firmware-crystalhd libfftw3-bin libfftw3-dev jackd2 libportaudio2 opus-tools pciutils pulseaudio libraw1394-doc librsvg2-bin lm-sensors serdi
  sndiod sordi speex nvidia-vdpau-driver nvidia-legacy-340xx-vdpau-driver nvidia-legacy-304xx-vdpau-driver
The following NEW packages will be installed:
  dbus ffmpeg fontconfig fontconfig-config fonts-dejavu-core i965-va-driver intel-media-va-driver libaacs0 libaom0 libapparmor1 libasound2 libasound2-data libass9 libasyncns0 libavc1394-0 libavcodec58 libavdevice58 libavfilter7 libavformat58 libavresample4
  libavutil56 libbdplus0 libbluray2 libbs2b0 libcaca0 libcairo2 libcdio-cdda2 libcdio-paranoia2 libcdio18 libchromaprint1 libcodec2-0.8.1 libcroco3 libcrystalhd3 libdatrie1 libdbus-1-3 libdc1394-22 libdrm-amdgpu1 libdrm-common libdrm-intel1 libdrm-nouveau2
  libdrm-radeon1 libdrm2 libelf1 libfftw3-double3 libflac8 libflite1 libfontconfig1 libfreetype6 libfribidi0 libgdk-pixbuf2.0-0 libgdk-pixbuf2.0-bin libgdk-pixbuf2.0-common libgl1 libgl1-mesa-dri libglapi-mesa libglib2.0-0 libglib2.0-data libglvnd0 libglx-mesa0
  libglx0 libgme0 libgomp1 libgraphite2-3 libgsm1 libharfbuzz0b libice6 libiec61883-0 libigdgmm5 libjack-jackd2-0 libjbig0 libjpeg62-turbo liblilv-0-0 libmp3lame0 libmpg123-0 libmysofa0 libnorm1 libnuma1 libogg0 libopenal-data libopenal1 libopenjp2-7 libopenmpt0
  libopus0 libpango-1.0-0 libpangocairo-1.0-0 libpangoft2-1.0-0 libpciaccess0 libpgm-5.2-0 libpixman-1-0 libpng16-16 libpostproc55 libpulse0 libraw1394-11 librsvg2-2 librsvg2-common librubberband2 libsamplerate0 libsdl2-2.0-0 libsensors-config libsensors5 libserd-0-0
  libshine3 libslang2 libsm6 libsnappy1v5 libsndfile1 libsndio7.0 libsodium23 libsord-0-0 libsoxr0 libspeex1 libsratom-0-0 libssh-gcrypt-4 libswresample3 libswscale5 libthai-data libthai0 libtheora0 libtiff5 libtwolame0 libusb-1.0-0 libva-drm2 libva-x11-2 libva2
  libvdpau-va-gl1 libvdpau1 libvidstab1.1 libvorbis0a libvorbisenc2 libvorbisfile3 libvpx5 libwavpack1 libwayland-client0 libwayland-cursor0 libwayland-egl1 libwebp6 libwebpmux3 libx11-6 libx11-data libx11-xcb1 libx264-155 libx265-165 libxau6 libxcb-dri2-0
  libxcb-dri3-0 libxcb-glx0 libxcb-present0 libxcb-render0 libxcb-shape0 libxcb-shm0 libxcb-sync1 libxcb-xfixes0 libxcb1 libxcursor1 libxdamage1 libxdmcp6 libxext6 libxfixes3 libxi6 libxinerama1 libxkbcommon0 libxrandr2 libxrender1 libxshmfence1 libxss1 libxtst6
  libxv1 libxvidcore4 libxxf86vm1 libzmq5 libzvbi-common libzvbi0 mesa-va-drivers mesa-vdpau-drivers shared-mime-info va-driver-all vdpau-driver-all x11-common xdg-user-dirs xkb-data
0 upgraded, 180 newly installed, 0 to remove and 0 not upgraded.
Need to get 65.5 MB of archives.
After this operation, 367 MB of additional disk space will be used.
```

Reply:
Also, apt leaves quite some cache lying around after an update. It is best practice to remove the cache in the same layer / command (see above in the Dockerfile).

Reply:
Hi Alexander, I did a comparsion: the sizes are 2.37G vs 2.23G. Is that 0.14G acceptable or do you have other alternative suggestion？

Reply:
Could you please share the method how you came to these values?
I guess it's two `make docker-build` with the size extracted from `docker image ls`?
There are some things to consider when talking about the docker image size:
- The interesting comparison is the "light" image, because it's the one which is pushed to `localstack/localstack`.
  - It can be built with `make docker-build-light`.
  - For me, the analysis looks as follows:
    ```
    # Build the light image without ffmpeg (without the changes on the PR)
    make docker-build-light
    docker tag localstack/localstack-light localstack/localstack-light-wo-ffmpeg
    # Get size of build with ffmpeg in bytes
    docker image inspect localstack/localstack-light-wo-ffmpeg | jq .[0].Size
    > 1864664703
    # Add the ffmpeg to the apt install / apply changes in the PR
    make docker-build-light
    # Get size of build with ffmpeg in bytes
    docker image inspect localstack/localstack-light | jq .[0].Size
    > 1993386742
    ```
  - So the addition of ffmpeg results in an increase of the uncompressed image size of 128722039 bytes or 122,76 MB.
  - (My results are a bit lower than yours because I added the ffmpeg install call to the apt install call in the base stage, where the cache is removed afterwards. Your `apt` call doesn't remove the apt cache.)
- The most interesting metric is the compressed image size (the one shown on DockerHub / the amount of data that needs to be downloaded).
  - I don't know of a way to reliably determine the compressed image size without pushing it to a registry.

While the 123 MB (uncompressed) seem like not too much, it's still an increase of 7% in image size just for this change. Since we have this use-case a lot, we try to tackle these problems by downloading the necessary software on demand (and cache it in the `var_libs` folder). In order to do that, we would need to have a statically linked ffmpeg build (assuming there are no issues with the license).
Do you know if there is a public downloadable statically linked build of ffmpeg for linux/arm64 and linux/amd64?

Reply:
Yes, my comparing process is pretty much like yours. I got your idea and I will try to find a statically liked build of ffmpeg. Thanks!

Reply:
I found this:
`https://github.com/jrottenberg/ffmpeg/blob/main/docker-images/5.1/ubuntu1804/Dockerfile`
Their ubuntu ffmpeg images is about 102MB which could potentially save us some space we compile the ffmpeg just like them. But there are two problems. 1. It doesn't support arm64. 2. It would drastically decrease the readability of our dockerfile. How do you think about it

Reply:
This would still add it directly to the docker image, instead of downloading a statically linked binary on demand when the transcribe service is used.
The best solution would be to find a publicly available download for both archs, which can be downloaded using a `PackageInstaller`, similar to the download of `kinesis-mock` here: https://github.com/localstack/localstack/blob/231547f3ce14e6d6ca2a5787f4204840c33383c7/localstack/services/kinesis/packages.py#L26

Reply:
Hi Alexander, I found compiled builds of [ffmpeg](https://twitter.com/ffmpeg/status/1040386019722440704). And I changed the code based on your suggestion. Right now, we don't need to worry about the docker image size. There is a function which can download ffmpeg binary on the fly.

Reply:
Hi Alexander, do you know how to solve that CLA issue? I am using a different computer. That bot seems recognize me  as other people.